packages/flutter_tools/test/commands.shard/permeable/analyze_test.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter_tools/src/base/common.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/analyze.dart';
+import 'package:flutter_tools/src/dart/pub.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+final Generator _kNoColorTerminalPlatform = () => FakePlatform.fromPlatform(const LocalPlatform())..stdoutSupportsAnsi = false;
+final Map<Type, Generator> noColorTerminalOverride = <Type, Generator>{
+  Platform: _kNoColorTerminalPlatform,
+};
+
+void main() {
+  group('analyze', () {
+    Directory tempDir;
+    String projectPath;
+
+    setUpAll(() {
+      Cache.disableLocking();
+      tempDir = fs.systemTempDirectory.createTempSync('flutter_analyze.').absolute;
+      projectPath = fs.path.join(tempDir.path, 'flutter_project');
+    });
+
+    tearDownAll(() {
+      tryToDelete(tempDir);
+    });
+
+    testUsingContext('--flutter-repo should run update-package first', () async {
+      await runCommand(
+        command: AnalyzeCommand(workingDirectory: fs.directory(projectPath)),
+        arguments: <String>['analyze', '--flutter-repo'],
+        statusTextContains: <String>['Running "flutter pub get" in automated_tests...'],
+      );
+    }, overrides: <Type, Generator>{
+      Pub: () => const Pub(),
```

Comment:
I think you should mock out `ProcessUtils` in the context overrides, and instead of grepping the output you can put expectations that it's invoking `flutter update-packages`. Then this test can be moved from `commands.shard/permeable/analyze_test.dart` to `commands.shard/hermetic/analyze_test.dart`.

This will make your test run a LOT faster, and it will also be more consistent. Generally speaking, we try not to call out to any external commands in our tests, because it leads to flakiness and ends up testing things outside our control (such as pub being able to download packages).